src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,20 +1,102 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Arrays;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final String NULL_EXPRESSION = "null";
+    private static final int MAX_RANGE = 10;
+    private int size;
+    private final K[] arrayOfKeys = (K[]) new Object[MAX_RANGE];
+    private final V[] arrayOfValues = (V[]) new Object[MAX_RANGE];
+    private boolean nullWasCreatedOnce = false;
 
     @Override
     public void put(K key, V value) {
+        boolean isUnique = true;
+        for (int i = 0; i < arrayOfKeys.length; i++) {
+            if (arrayOfKeys[i] != null && key != null && arrayOfKeys[i].equals(key.toString())) {
+                arrayOfValues[i] = (V) (value != null ? value.toString() : NULL_EXPRESSION);
+                isUnique = false;
+            } else if (key == null && arrayOfKeys[i] == null && isUnique) {
+                arrayOfValues[i] = (V) (value != null ? value.toString() : NULL_EXPRESSION);
+                isUnique = false;
+                if (nullWasCreatedOnce == false) {
+                    size++;
+                    nullWasCreatedOnce = true;
+                }
+            }
+        }
+        if (isUnique) {
+            if (key == null) {
+                key = (K) NULL_EXPRESSION;
+            }
+            arrayOfKeys[size] = (K) key.toString();
+            arrayOfValues[size] = (V) (value != null ? value.toString() : NULL_EXPRESSION);
+            size++;
+        }
```

Comment:
```suggestion
        arrayOfKeys[size] = key;
        arrayOfValues[size] = value;
        size ++;
```
the same with get method